app/controllers/webhooks_controller.rb

```diff
@@ -15,14 +15,15 @@ def sendgrid
     events.each do |event|
       message_id = Email::MessageIdService.message_id_clean((event["smtp-id"] || ""))
       to_address = event["email"]
+      error_code = event["status"]
       if event["event"] == "bounce"
-        if event["status"]["4."]
-          process_bounce(message_id, to_address, SiteSetting.soft_bounce_score)
+        if error_code["4."]
```

Comment:
Would be nice to know that "4." means "Persistent Transient Failure"

Reply:
Yes good point since we also do this in the receiver `https://github.com/discourse/discourse/blob/3f0d63f98d815a32f58d6e4eed3ed1ab76943a9b/lib/email/receiver.rb#L254-L254`. Will put this and 5. and 2. in a const that we can check against!